docs/api/index.md

```diff
@@ -0,0 +1,180 @@
+# API Reference
+
+Welcome to the Kitbag query API reference. This section provides detailed documentation for all functions, types, and interfaces available in the library.
+
+## Core Functions
+
+### createQueryClient()
+
+Creates a new query client instance with optional configuration.
+
+**Type Signature:**
+```ts
+function createQueryClient(options?: ClientOptions): {
+  query: QueryClient
+  useQuery: <T>(query: Query<T>, options?: UseQueryOptions<T>) => QueryComposition<T>
+}
+```
+
+**Parameters:**
+- `options` (optional): Configuration options for the query client
+
+**Returns:**
+- An object containing `query` and `useQuery` functions
+
+### query()
+
+Creates a query function for data fetching.
+
+**Type Signature:**
+```ts
+function query<TParams extends any[], TData>(
+  key: string,
+  fetcher: (...params: TParams) => Promise<TData>,
+  options?: QueryOptions
+): QueryFunction<TParams, TData>
```

Comment:
Incorrect example

Reply:
I'll comment this out and revisit in future PR